### PR TITLE
Add integration test for Redis unavailability.

### PR DESCRIPTION
**Pull request by vadimberezniker**
This test starts a real app & executor, waits for a remote execution to
start, restarts the Redis shard which contains the task information and
verifies that the execution can still finish successfully.

---

**Version bump**: None

### CODE REVIEW

**Review by bduffany — APPROVED** [review]
LGTM modulo test failure -- this is a really nice test to have in place :+1:

File: enterprise/server/testutil/testredis/testredis.go [hunk 1]
```diff
@@ -115,6 +129,24 @@ func Start(t testing.TB) *Handle {
 	return handle
 }
 
+// StartTCP spawns a Redis server listening on a TCP port.
+// Most uses should prefer the Start function. This function should be used for cases where a unix socket is not
```
Comment (tylerwilliams) [inline enterprise/server/testutil/testredis/testredis.go:133]: nit: 80 chars for comments

File: enterprise/server/test/integration/remote_execution/remote_execution_test.go [hunk 2]
```diff
@@ -1015,3 +1021,68 @@ func TestCommandWithMissingInputRootDigest(t *testing.T) {
...
+					require.FailNow(t, "multiple redis shards contain task information")
+				}
+				victimShard = shard
+			}
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	require.NotNil(t, victimShard, "could not find victim shard")
+
+	// Restart the shard containing task information and verify that the Bazel invocation can still finish successfully.
```
Comment (tylerwilliams) [inline enterprise/server/test/integration/remote_execution/remote_execution_test.go:1077]: 80

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Comment by vadimberezniker** [comment]
> LGTM modulo test failure -- this is a really nice test to have in place +1

This test depends on Zoey's changes in https://github.com/buildbuddy-io/buildbuddy/pull/1711
Without the flag fixes, the --remote_execution.sharded_redis.shards flags have no effect.

File: enterprise/server/testutil/testredis/testredis.go (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/server/testutil/testredis/testredis.go:133]: Done.

File: enterprise/server/test/integration/remote_execution/remote_execution_test.go (hunk 2 above)
Comment (vadimberezniker) [reply enterprise/server/test/integration/remote_execution/remote_execution_test.go]: Done.

**Pull request merged**